pilot/pkg/xds/auth.go

```diff
@@ -57,15 +58,18 @@ func (s *DiscoveryServer) authenticate(ctx context.Context) ([]string, error) {
 		return nil, nil
 	}
 	authFailMsgs := []string{}
+	authContext := security.NewAuthContext(ctx)
 	for _, authn := range s.Authenticators {
-		u, err := authn.Authenticate(ctx)
-		// If one authenticator passes, return
+		u, err := authn.Authenticate(authContext)
```

Comment:
This is one change - we are now running all the authenticators.

Reply:
I expect xfcc to be last, and to swap the identity set by the previous ones if it is trusted.

This is the kind of behavior that should have a proper doc and be reviewed for security, but at least add extensive comments.

Reply:
Mtls is first and xfcc is last.  It appends to the list of identities that previous ones have already added. 

> Not sure how this works - first wins?

Any one wins. The identities are merged together.

Reply:
Not sure the method signature is right - no need to return 'u' - authenticators should modify the context.

Also err is not clear how to use - maybe have a field in the context with error conditions. I believe the current model is pretty bad - it is not an error to not have a JWT or mTLS client cert - but it is an error to have an expired JWT or mTLS cert, or invalid XFCC format. A real error in one of the authenticators should abort, for example if mTLS is good and XFCC is bad we shouldn't let the mTLS identity stand.

Reply:
> Any one wins. The identities are merged together.

This doesn't seem correct. Example: we have `admin` identity with `XFFC=low-priv`. Now the low-priv client has admin access if they want?

Reply:
This is not true any more. XFCC is added at the end and will only authenticate based xfcc header if the client connection is part of trusted networks defined in env var or local host. There is no merging of identities. Those were old comments.